docs/NETWORK-MODELS.md

```diff
@@ -0,0 +1,41 @@
+# Network Models
+
+What are the network models that EVE supports for Edge Container workloads, and how are they implemented?
+
+The first half of this document describes _what_ those network models are and how they impact Edge Container communications. The second half describes _how_ those models are implemented.
+
+## Models
+
+Fundamentally, each Edge Container needs to communicate with services outside itself. Those can be one or more of:
+
+* other Edge Container on the same device
+* the network to which the device is attached
+* other Edge Container on one or more other devices
+* a network (VPC) in the public cloud
+* no networking at all
+
+It is possible to define all network requirements using communications permissions model. For example:
+
+* Edge Container-to-EdgeContainer: mesh (between devices) or switch (same device)
+* Edge Container-to-cloud: VPN
+* Edge Container-to-host-network-NAT: switche
+* Edge Container-to-host-network-transparent: bridge
+
+In principle, a Edge Container definition does not define any of its methodologies, networks or technologies. It simply defines what it should be able to connect to. The Controller then is responsible for translating those networking permissions to implementations passed on to EVE.
+
+## Implementations
+
+EVE currently supports four primary networking implementations - in addition to "no network" - to support the above models. These are:
+
+* Bridged - EVE sets up a transparent L2 bridge between the Edge Containers and the underlying device network. The responsibility for IP address management (IPAM) is handed off to the underlying fabric.
+* Switched - EVE sets up a switch between the Edge Containers and the underlying device network. EVE provides IPAM to the Edge Containers, and NAT between the Edge Containers and the device network.
```

Comment:
Doesn't exist.

There is a "local" which includes IPAM (plus DNS etc) where the NAT to the outside world is optional. The NAT is selected by specifying an adapter (XXX have to check that the term is in the API for the network instance)